pkg/odo/cli/url/create.go

```diff
@@ -118,10 +122,18 @@ func (o *URLCreateOptions) Complete(name string, cmd *cobra.Command, args []stri
 			return err
 		}
 
-		if len(args) == 0 {
-			o.urlName = url.GetURLName(componentName, o.componentPort)
+		if pushtarget.IsPushTargetDocker() {
+			o.urlName = "local-" + url.GetURLName(componentName, o.componentPort)
+			o.exposedPort, err = url.GetValidExposedPortNumber(o.exposedPort)
+			if err != nil {
+				return err
+			}
 		} else {
-			o.urlName = args[0]
+			if len(args) == 0 {
```

Comment:
no need for `else` statement.. you can do:

```go
		if pushtarget.IsPushTargetDocker() {
			o.urlName = "local-" + url.GetURLName(componentName, o.componentPort)
			o.exposedPort, err = url.GetValidExposedPortNumber(o.exposedPort)
			if err != nil {
				return err
			}
		} else if len(args) == 0 {
				o.urlName = url.GetURLName(componentName, o.componentPort)
		} else {
				o.urlName = args[0]
	       }
```

Reply:
updated